odoo/service/server.py

```diff
@@ -63,6 +64,14 @@
 
 SLEEP_INTERVAL = 60     # 1 min
 
+
+# A global-ish object, each thread/worker uses its own
+thread_local = threading.local()
```

Comment:
Why do you use thread local here?
For all other attributes, we store them directly on the thread.